Remove incentives from moderator participants.

eschlange opened the pull request:
* Remove incentive creation in all cases for moderators.
* Update specs to expect no moderator participant.
* Add migration that removes existing participant incentives form mods.
* Update rubocop settings for method and line length.

[#118403907]

eschlange commented:
@michaelwehrley @ericcf If you have a little time can you take a look at a data migration that removes moderator incentives and behaviors? I just want another set of eyes looking at the logic before I run it on production: sunnyside/db/migrate/20160428165116_delete_participant_incentives_and_participant_behaviors_from_moderators.rb

michaelwehrley commented on app/models/sunnyside/incentive.rb:
```diff
@@ -46,12 +47,14 @@ def self.create_membership_incentives(membership)
+          is_notification_sent: false,
+          is_complete: false
+        )
+      end
```
@eschlange could please you make this a sentry error instead and is this method the reason for the increase in MethodLength?

eschlange replied:
Yes, I'll convert these logging statements to use Raven.

eschlange replied:
Also abstracted out the logging to revert rubocop methodlength change.

michaelwehrley commented on db/migrate/20160428165116_delete_participant_incentives_and_participant_behaviors_from_moderators.rb:
```diff
@@ -0,0 +1,8 @@
+class DeleteParticipantIncentivesAndParticipantBehaviorsFromModerators < ActiveRecord::Migration
+  def change
+    Participant.where(is_admin: true).each do |admin_participant|
```
seems reasonable.  I would just try it with production db locally first.

eschlange replied:
Looks good on the prod DB copied in locally, no errors. I'll make sure to ensure the stability of the migration changes on Staging as well.

ericcf replied:
@michaelwehrley @eschlange This seems like a perfect use case for a migration model, don't you think? See https://github.com/cbitstech/conemo_dashboard/blob/9109c9877bf1ca574a6887a81443f69418f00978/db/migrate/20160426203216_update_supervisor_note_reference.rb and http://blog.testdouble.com/posts/2014-11-04-healthy-migration-habits.html.

michaelwehrley commented:
Nice job.  Just a few comments.  Thanks.

ericcf commented on app/models/sunnyside/incentive.rb:
```diff
@@ -31,30 +31,39 @@ def self.create_membership_incentives(membership)
+      unless participant.is_admin
+        ParticipantIncentive.create!(
+          participant: participant,
+          incentive: self,
```
A more idiomatic Rails way (I would argue) to do this would be to call `participant_incentives.create!` - see http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#module-ActiveRecord::Associations::ClassMethods-label-Collection+associations+-28one-to-many+-2F+many-to-many-29

eschlange commented:
@ericcf Please take a look at my changes to the migration. I modified it to use migration models. I also removed the method that you mentioned and simply replaced its usage with create!.

ericcf commented:
Looks better to me. @michaelwehrley do you concur?

michaelwehrley commented on db/migrate/20160428165116_delete_participant_incentives_and_participant_behaviors_from_moderators.rb:
```diff
@@ -0,0 +1,21 @@
+class DeleteParticipantIncentivesAndParticipantBehaviorsFromModerators < ActiveRecord::Migration
+  class MigrationParticipantBehavior < ActiveRecord::Base
+    self.table_name = :participant_behaviors
```
:oh_yeah!

michaelwehrley commented:
👀 great!  Thank for the updates!
